yt/yt/server/queue_agent/queue_static_table_exporter.cpp

```diff
@@ -773,42 +746,87 @@ TQueueExportProfilingCounters::TQueueExportProfilingCounters(const TProfiler& pr
 
 TQueueExporter::TQueueExporter(
     TString exportName,
+    TCrossClusterReference queue,
+    const NQueueClient::TQueueStaticExportConfig& config,
     TClientDirectoryPtr clientDirectory,
     IInvokerPtr invoker,
     IAlertCollectorPtr alertCollector,
     const TProfiler& queueProfiler,
     const TLogger& logger)
     : ExportName_(std::move(exportName))
+    , Queue_(std::move(queue))
+    , Config_(config)
+    , ExportProgress_(New<TQueueExportProgress>())
     , ClientDirectory_(std::move(clientDirectory))
     , Invoker_(std::move(invoker))
     , AlertCollector_(std::move(alertCollector))
     , ProfilingCounters_(New<TQueueExportProfilingCounters>(queueProfiler.WithPrefix("/static_export").WithTag("export_name", ExportName_)))
     , Logger(logger.WithTag("ExportName: %v", ExportName_))
 { }
 
-TFuture<void> TQueueExporter::RunExportIteration(
-    const TCrossClusterReference& queue,
-    const TQueueStaticExportConfig& exportConfig)
+TFuture<void> TQueueExporter::RunExportIteration()
 {
+    auto config = GetConfig();
+
     auto exportTask = New<TQueueExportTask>(
-        ClientDirectory_->GetClientOrThrow(queue.Cluster),
+        ClientDirectory_->GetClientOrThrow(Queue_.Cluster),
         Invoker_,
         ProfilingCounters_,
-        queue.Path,
-        exportConfig,
+        Queue_.Path,
+        config,
         Logger);
+
     auto exportTaskFuture = exportTask->Run();
-    exportTaskFuture.Subscribe(BIND([this, this_ = MakeStrong(this)] (const TError& error) {
-        if (!error.IsOK()) {
+
+    exportTaskFuture.Subscribe(BIND([
+        this,
+        this_ = MakeStrong(this),
+        config = std::move(config)
+    ] (const TErrorOr<TQueueExportProgressPtr>& exportProgress) {
+        if (!exportProgress.IsOK()) {
             AlertCollector_->StageAlert(CreateAlert(
                 NAlerts::EErrorCode::QueueAgentQueueControllerStaticExportFailed,
                 "Failed to perform static export for queue",
                 /*tags*/ {{"export_name", ExportName_}},
-                error));
+                /*error*/ exportProgress));
+        } else {
+            // NB(apachee): This assumes that queue exporter owns the returned pointer,
+            // and it shouldn't be modified by anyone else.
+            auto nextExportProgress = exportProgress.Value();
+
+            auto guard = Guard(Lock_);
+
+            if (config == Config_ && nextExportProgress->LastExportedFragmentUnixTs > ExportProgress_->LastExportedFragmentUnixTs) {
```

Comment:
Well, the first one is there to prevent race condition, when export is pending, config changes, and the change involves `ExportDirectory`. In this case after export has finished the code without this condition would update export progress even though `ExportDirectory` of the export and the one in the config don't match. It should probably be replaced with `config.ExportDirectory != Config_.ExportDirectory` though, because not all of the changes cause an issue.

The second condition was there, because I didn't want to make any assumptions about whether the export that started later than another one would always finish later too. It is not as much of an issue as the previous one, so this condition is not as important, and I can get rid of it.

Reply:
Ah, makes sense. Seems worthy of a comment, and I agree that comparing `ExportDirectory` explicitly is better. 

As for the second condition, we never run more than export at once, so it is not an issue.